presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchClient.java

```diff
@@ -0,0 +1,264 @@
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.VarbinaryType.VARBINARY;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public class ElasticsearchClient
+{
+    private static final Logger log = Logger.get(ElasticsearchClient.class);
+
+    private final ElasticsearchConnectorConfig config;
+    private final Map<SchemaTableName, ElasticsearchTableDescription> tableDescriptions;
+
+    private Map<String, Client> clients;
+
+    @Inject
+    public ElasticsearchClient(ElasticsearchConnectorConfig elasticsearchConfig, Map<SchemaTableName, ElasticsearchTableDescription> descriptions)
+            throws IOException
+    {
+        config = elasticsearchConfig;
+        tableDescriptions = descriptions;
+        createClients();
+    }
+
+    public ElasticsearchConnectorConfig getConfig()
+    {
+        return config;
+    }
+
+    public ImmutableMap<String, Client> getClients()
+    {
+        return ImmutableMap.copyOf(clients);
+    }
+
+    public List<String> listSchemas()
+    {
+        ImmutableSet.Builder<String> builder = ImmutableSet.builder();
+        for (SchemaTableName tableName : tableDescriptions.keySet()) {
+            builder.add(tableName.getSchemaName());
+        }
+        return ImmutableList.copyOf(builder.build());
+    }
+
+    public List<SchemaTableName> listTables(String schemaNameOrNull)
+    {
+        ImmutableList.Builder<SchemaTableName> builder = ImmutableList.builder();
+        for (SchemaTableName tableName : tableDescriptions.keySet()) {
+            if (schemaNameOrNull == null || tableName.getSchemaName().equals(schemaNameOrNull)) {
+                builder.add(tableName);
+            }
+        }
+
+        return builder.build();
+    }
+
+    public ElasticsearchTableDescription getTable(String schemaName, String tableName)
+    {
+        requireNonNull(schemaName, "schemaName is null");
+        requireNonNull(tableName, "tableName is null");
+        try {
+            for (Map.Entry<SchemaTableName, ElasticsearchTableDescription> entry : tableDescriptions.entrySet()) {
+                buildColumns(entry.getKey(), entry.getValue());
+            }
+        }
+        catch (Exception e) {
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, e.getMessage());
```

Comment:
Instead of relying on generic error codes, you can define a new set of codes specific to the elastic search connector (see  `HiveErrorCode`, `KafkaErrorCode`, etc. for examples).

Also, instead of a catch all statement here you should throw appropriate codes on specific exceptions, that will be much more useful to the user when an error occurs.